### PR TITLE
Make readable stores available to all services

### PR DESCRIPTION
**Pull request by netopyr**
This PR contains the first set of changes that will make readable stores available to the pre-handle and query handlers of all services.

### CODE REVIEW

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java [hunk 1]
```diff
@@ -385,4 +414,42 @@ public String toString() {
...
+    private static class ReadableStoreFactoryHack implements ReadableStoreFactory {
+        private final AccountAccess accountAccess;
+
+        private ReadableStoreFactoryHack(@NonNull final AccountAccess accountAccess) {
+            this.accountAccess = accountAccess;
+        }
+
+        @Override
+        public <C> C createStore(final Class<C> storeInterface) {
+            if (storeInterface == AccountAccess.class) {
```
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java:449]: null check & annotation

**Review by hendrikebbers — APPROVED** [review]

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java [hunk 2]
```diff
@@ -385,4 +414,42 @@ public String toString() {
...
+         * @param <C> Interface class for a Store
+         * @return An implementation of the provided store interface
+         * @throws IllegalArgumentException if the storeInterface class provided is unknown to the app
+         * @throws NullPointerException if {@code clazz} is {@code null}
+         */
+        @NonNull
+        <C> C createStore(@NonNull final Class<C> storeInterface) throws IllegalArgumentException;
+    }
+
+    private static class ReadableStoreFactoryHack implements ReadableStoreFactory {
```
[the remark this answers is not in the document]
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java:440]: Sure, though I think the risk that somebody misuses a private class that is called `Hack` are rather small. ;)

[the remark this answers is not in the document]
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java:441]: Done

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java (hunk 1 above)
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java:452]: done

**Review by hendrikebbers — APPROVED** [review]

**Pull request merged**